src/QirRuntime/lib/QIR/intrinsicsMath.cpp

```diff
@@ -28,4 +39,38 @@ double quantum__qis__arctan2__body(double y, double x)
     return std::atan2(y, x);        // https://en.cppreference.com/w/cpp/numeric/math/atan2
 }
 
+int64_t quantum__qis__drawrandomint__body(int64_t minimum, int64_t maximum)
+{
+    if(minimum > maximum)
+    {
+        quantum__rt__fail(quantum__rt__string_create(Quantum::Qis::Internal::excStrDrawRandomInt));
+    }
+
+    // https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution
+    // https://en.cppreference.com/w/cpp/numeric/random
+    thread_local static std::mt19937_64 gen(seedIsRandom
+                                                ? std::random_device()() :  // Default
+                                                0);                         // For test purposes only.
+
+    return std::uniform_int_distribution<int64_t>(minimum, maximum)(gen);
+}
+
 }   // extern "C"
+
+namespace   // Visible in this translation unit only.
+{
+    thread_local bool seedIsRandom = true;
+}
+
+// For test purposes only:
+namespace Quantum { namespace Qis { namespace Internal 
+{
+
+char const excStrDrawRandomInt[] = "Invalid Argument: minimum > maximum for DrawRandomInt()";
+
+void UseRandomSeed(bool random)
```

Comment:
`SetRandomSeed`?

Reply:
Oh... ;-) That's what is standing behind UseFoo() that I saw earlier... OK. Nice to know.
Renaming `UseRandomSeed()` to `RandomizeSeed()`.